code/modules/antagonists/traitor/IAA/internal_affairs.dm

```diff
@@ -66,13 +68,13 @@
 		linked_alert.icon_state = "pinondirect"
 	else
 		linked_alert.setDir(get_dir(here, there))
-		switch(get_dist(here, there))
-			if(1 to 8)
-				linked_alert.icon_state = "pinonclose"
-			if(9 to 16)
-				linked_alert.icon_state = "pinonmedium"
-			if(16 to INFINITY)
-				linked_alert.icon_state = "pinonfar"
+		var/dist = (get_dist(here, there))
```

Comment:
you can't use the switch here?

```DM
		var/dist = (get_dist(here, there))
		switch(dist)
		    if(1 to range_mid)
                ...
		    if((range_mid+1) to range_far)
                ...
		    else
                ...
```

I never tried to do it like that and cannot test atm so dunno if it will work :P

Reply:
Switch only works with constants (according to byond docs and it not working properly when i tested it with variables)

Reply:
lame and g*y